wp-tevko-responsive-images.php

```diff
@@ -301,31 +303,30 @@ function tevkori_get_src_sizes( $id, $size = 'thumbnail' ) {
 }
 
 /**
- * Filter for extending image tag to include srcset attribute.
+ * Filter for the_content to add sizes and srcset attributes to images.
+ *
+ * @since 3.0
  *
- * @see images_send_to_editor
- * @return string HTML for image.
+ * @param string $content The raw post content to be filtered.
  */
-function tevkori_extend_image_tag( $html, $id, $caption, $title, $align, $url, $size, $alt ) {
-	add_filter( 'editor_max_image_size', 'tevkori_editor_image_size' );
-
-	// Get the srcset attribute.
-	$srcset = tevkori_get_srcset_string( $id, $size );
-
-	remove_filter( 'editor_max_image_size', 'tevkori_editor_image_size' );
+function tevkori_filter_content_images( $content ) {
+	preg_match_all( '/src="([^"]*)"/', $content, $matches );
 
-	if ( $srcset ) {
+	$urls = array_pop( $matches );
 
-		// Build the data-sizes attribute if sizes were returned.
-		$sizes = tevkori_get_sizes( $id, $size );
-		$sizes = $sizes ? 'data-sizes="' . $sizes . '"' : '';
+	$attachments = tevkori_attachment_urls_to_loop( $urls );
```

Comment:
The URLs that we pass to the function contain the size suffix (eg. -300x200) if present. I think we have to strip that like we do in https://github.com/ResponsiveImagesCG/wp-tevko-responsive-images/pull/170/files#diff-517abe0a5b7ee2c55c6b5659d0c9350eR52 but would be nice if we don't have to do it at two places.